contracts/StaticMarket.sol

```diff
@@ -0,0 +1,100 @@
+/*
+
+  << Static Market contract >>
+
+*/
+
+pragma solidity 0.7.5;
+
+import "./static/StaticUtil.sol";
+
+/**
+ * @title StaticMarket
+ * @author Wyvern Protocol Developers
+ */
+contract StaticMarket is StaticUtil {
+
+    string public constant name = "Static Market";
+
+	constructor (address atomicizerAddress)
+		public
+	{
+		atomicizer = atomicizerAddress;
+	}
+
+	function anyERC1155ForERC20(bytes memory extra,
+		address[7] memory addresses, AuthenticatedProxy.HowToCall[2] memory howToCalls, uint[6] memory uints,
+		bytes memory data, bytes memory counterdata)
+		public
+		pure
+		returns (uint)
+	{
+		require(uints[0] == 0,"anyERC1155ForERC20: Zero value required");
+		require(howToCalls[0] == AuthenticatedProxy.HowToCall.Call, "anyERC1155ForERC20: call must be a direct call");
+
+		(uint8 direction, address[2] memory tokenGiveGet, uint256[2] memory nftIdAndPrice) = abi.decode(extra, (uint8, address[2], uint256[2]));
+
+		require(nftIdAndPrice[1] > 0,"anyERC1155ForERC20: ERC1155 price must be larger than zero");
+
+		uint256 new_fill;
+
+		if (direction == 0){
```

Comment:
Why `direction` instead of using two separate static call functions? This is fine but the latter is probably (ever so slightly) more efficient.

Reply:
(you could even have this kind of abstraction but have two helper functions, theoretically the compiler should optimise)

Reply:
The reasoning was to make constructing the static call as easy as possible in both directions. You only toggle the direction. But yes, can also make it two separate calls. What is your recommendation / preference?

Reply:
Actually I am refactoring this now. Another partial static call I am working on led to new insights.